app/controllers/shelters_controller.rb

```diff
@@ -18,6 +18,27 @@ def create
     @shelter.state = params[:state]
     @shelter.zip = params[:zip]
     @shelter.save
+
     redirect_to '/shelters'
   end
+
+  def edit
+    @shelter = Shelter.find(params[:id])
+  end
+
+  def update
+    @shelter = Shelter.find(params[:id])
+    @shelter.update(shelter_params)
+
+    redirect_to "/shelters/#{@shelter.id}"
+  end
+
+  def destroy
+    Shelter.destroy(params[:id])
+    redirect_to "/shelters"
+  end
+
+  def shelter_params
```

Comment:
This shelter param could probably be used in the create method as well. Will note for refactoring